qbraid/runtime/job.py

```diff
@@ -85,6 +86,61 @@ def wait_for_final_state(self, timeout: Optional[int] = None, poll_interval: int
                 raise TimeoutError(f"Timeout while waiting for job {self.id}.")
             sleep(poll_interval)
 
+    async def _wait_for_final_state(
+            self, timeout: Optional[int] = None, poll_interval: int = 5
+    ) -> None:
+        """Asynchronously wait for the job to reach a terminal state (e.g., COMPLETED, FAILED).
+
+        This non-blocking method uses asyncio to periodically poll the job's status,
+        allowing other coroutines to run while waiting. It is especially useful in
+        asynchronous applications where blocking the event loop is undesirable.
+
+        Args:
+            timeout: Seconds to wait for the job. If ``None``, wait indefinitely.
+            poll_interval: Seconds between queries. Defaults to 5 seconds.
+
+        Raises:
+            JobStateError: If the job does not reach a final state before the specified timeout.
+
+        """
+        start_time = time()
+        while not self.is_terminal_state():
+            elapsed_time = time() - start_time
+            if timeout is not None and elapsed_time >= timeout:
+                raise TimeoutError(f"Timeout while waiting for job {self.id}.")
+            await asyncio.sleep(poll_interval)
+
+    async def async_result(
+            self, timeout: Optional[int] = None, poll_interval: int = 5
+    ) -> qbraid.runtime.Result[ResultDataType]:
+        """
+        Asynchronously wait for the job to reach a final state and return the result.
+
+        This method provides a non-blocking way to poll the job status using asyncio.
+        It allows developers to `await` the job without blocking the event loop,
+        making it suitable for integration in async applications and pipelines.
+
+        Parameters
+        ----------
+        timeout : int, optional
+            Maximum number of seconds to wait for the job. If ``None``, waits indefinitely.
+
+        poll_interval : int, default=5
+            Number of seconds between status checks while waiting.
+
+        Raises
+        ------
+        JobStateError
+            If the job does not reach a final (terminal) state before the timeout expires.
+
+        Returns
+        -------
+        Result[ResultDataType]
+            The result object associated with the job, if successfully completed.
```

Comment:
Can you switch these docstrings to use the [Google style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)?